src/components/StakingVaults/hooks/useFetchOpporunities.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  avalancheChainId,
+  bchChainId,
+  bscChainId,
+  btcChainId,
+  cosmosChainId,
+  dogeChainId,
+  ethChainId,
+  fromAccountId,
+  ltcChainId,
+} from '@shapeshiftoss/caip'
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { setTimeoutAsync } from 'lib/utils'
+import { nftApi } from 'state/apis/nft/nftApi'
+import { zapper } from 'state/apis/zapper/zapperApi'
+import {
+  fetchAllOpportunitiesIdsByChainId,
+  fetchAllOpportunitiesMetadataByChainId,
+  fetchAllOpportunitiesUserDataByAccountId,
+} from 'state/slices/opportunitiesSlice/thunks'
+import {
+  selectPortfolioAccounts,
+  selectPortfolioAssetIds,
+  selectPortfolioLoadingStatus,
+  selectWalletAccountIds,
+} from 'state/slices/selectors'
+import { useAppDispatch } from 'state/store'
+
+export const useFetchOpportunities = () => {
+  const dispatch = useAppDispatch()
+  const portfolioLoadingStatus = useSelector(selectPortfolioLoadingStatus)
+  const requestedAccountIds = useSelector(selectWalletAccountIds)
+  const portfolioAssetIds = useSelector(selectPortfolioAssetIds)
+  const portfolioAccounts = useSelector(selectPortfolioAccounts)
+  const DynamicLpAssets = useFeatureFlag('DynamicLpAssets')
+
+  const [isLoading, setIsLoading] = useState(true)
+
+  // TODO: this needs to be split up into separate react-queries so we aren't refetching when moving around the app
```

Comment:
Not react-query queries currently, though such slices would be perfect contenders for react-queries, especially if we add persistence!

```suggestion
  // TODO: this needs to be split up into separate RTK queries so we aren't refetching when moving around the app
```

Reply:
@woodenfurniture comment suggestion still valid 🙏🏽

Reply:
Dangit, didnt push. Will do in the follow up with the other changes.